l1-contracts/contracts/state-transition/chain-deps/facets/Mailbox.sol

```diff
@@ -141,24 +157,30 @@ contract MailboxFacet is Base, IMailbox {
     }
 
     /// @notice Derives the price for L2 gas in ETH to be paid.
-    /// @param _l1GasPrice The gas price on L1.
+    /// @param _l1GasPrice The gas price on L1
     /// @param _gasPerPubdata The price for each pubdata byte in L2 gas
-    /// @return The price of L2 gas in ETH
+    /// @return The price of L2 gas in the base token
     function _deriveL2GasPrice(uint256 _l1GasPrice, uint256 _gasPerPubdata) internal view returns (uint256) {
         FeeParams memory feeParams = s.feeParams;
-
-        uint256 pubdataPriceETH;
+        require(s.baseTokenGasPriceMultiplierDenominator > 0, "Mailbox: baseTokenGasPriceDenominator not set");
+        uint256 l1GasPriceConverted = (_l1GasPrice * s.baseTokenGasPriceMultiplierNominator) /
+            s.baseTokenGasPriceMultiplierDenominator;
+        uint256 pubdataPriceBaseToken;
         if (feeParams.pubdataPricingMode == PubdataPricingMode.Rollup) {
-            pubdataPriceETH = L1_GAS_PER_PUBDATA_BYTE * _l1GasPrice;
+            // slither-disable-next-line divide-before-multiply
+            pubdataPriceBaseToken = L1_GAS_PER_PUBDATA_BYTE * l1GasPriceConverted;
         }
 
-        uint256 batchOverheadETH = uint256(feeParams.batchOverheadL1Gas) * _l1GasPrice;
-        uint256 fullPubdataPriceETH = pubdataPriceETH + batchOverheadETH / uint256(feeParams.maxPubdataPerBatch);
+        // slither-disable-next-line divide-before-multiply
+        uint256 batchOverheadBaseToken = uint256(feeParams.batchOverheadL1Gas) * l1GasPriceConverted;
+        uint256 fullPubdataPriceBaseToken = pubdataPriceBaseToken +
```

Comment:
We should put some eyes here. I'm not sure if we need to do something (this change is new).